Added AirCCPropertyListing.cs

CCianfloneDev opened the pull request:
Requesting DZ-FSDev to review code.

CCianfloneDev committed:
Added AirCCPropertyListing.cs

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+namespace AirCandC_api
+{
+    /// <summary>
+    /// Represents a rental property listing on Air c&c.
```
- [x] Air C&C

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        private int squareFootage;
+        private string address;
+        private RenterPackages rentTypeOffered;
+        private int priceForPackage;
```
- [x] Convert value-type fields to props.

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        private List<Amenities> amenities = new List<Amenities>();
+
+        /// <summary>
+        /// Initializes a listing on Air c&c
```
- [x] Air C&C

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        /// <summary>
+        /// Prints amenities associated with property.
+        /// </summary>
+        public void PrintAmenities()
```
- [x] Consider changing to `GetAmenitiesString() : string` after impl.

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        }
+
+        /// <summary>
+        /// Returns a string representation of the listing on Air c&c.
```
- [x] Air C&C

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        /// <param name="amenities">Amenities offered.</param>
+        public void SetAmenitiesOffered(List<Amenities> amenities)
+        {
+            this.amenities = amenities;
```
- [x] Consider setting a clone of the list to the field to uphold Encapsulation.

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+            {
+                Console.WriteLine($" {amenities} ");
+            }
+            if (amenities.Count <= 0)
```
- [x] Optimization: This should happen before the loop.

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,198 @@
+        /// <returns>Amenities offered.</returns>
+        public List<Amenities> GetAmenitiesOffered()
+        {
+            return this.amenities;
```
- [x] Consider returning a clone of the list to uphold Encapsulation.

CCianfloneDev replied:
Changed method to return clone of list to uphold encapsulation, however this method is not needed at this time, therefor I have commented it out for the time being.

DZ-FSDev reviewed (changes requested):
**Change Requests for AirCCPropertyListing.cs**

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Corrected documentation.

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Converted value-type fields to props.

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Corrected documentation

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Changed PrintAmenities method to a property, it's functioning as intended now.

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Updated GetAmenitiesOffered method to return clone of enum list, commented out as no functionality for said method at this moment.

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Changed SetAmenitiesOffered method to set the field to a clone of the list to uphold Encapsulation.

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,176 @@
+        /// <summary>
+        /// Gets price for rent type offered.
+        /// </summary>
+        public int PriceForPackage
+        {
+            get;
+            private set;
+        }
```
- [x] Currency amount should be changed to use `decimal`

DZ-FSDev commented on AirCAndC/AirCCPropertyListing.cs:
```diff
@@ -0,0 +1,176 @@
+        /// <param name="squareFootage">Square footage of property.</param>
+        /// <param name="address">Address associated with property.</param>
+        /// <param name="rentTypeOffered">Rent package associated with this listing (daily, weekly, monthly)</param>
+        /// <param name="priceForPackage">Price associated with renting package chosen.</param>
+        /// <param name="amenities">Amenities associated with this listing.</param>
+        public AirCCPropertyListing(int numberOfRooms, int numberOfBathrooms, int squareFootage, 
+            string address, RenterPackages rentTypeOffered, int priceForPackage, List<Amenities> amenities)
```
- [x] `priceForPackage` should be changed to the `decimal` tyoe

DZ-FSDev reviewed (changes requested):
**CR for #14**
Looks good! I missed this on my initial review:

CCianfloneDev committed:
Update AirCCPropertyListing.cs

Changed PriceForPackage prop from int value type to decimal type.

DZ-FSDev reviewed (approved):
Looks good!
